src/Analyzers/CSharp/CodeFixes/UseIndexOrRangeOperator/CSharpUseRangeOperatorCodeFixProvider.cs

```diff
@@ -110,12 +107,19 @@ private static ExpressionSyntax FixOne(Result result, SyntaxGenerator generator)
             if (result.MemberInfo.OverloadedMethodOpt == null)
             {
                 var argList = invocation.ArgumentList;
-                return ElementAccessExpression(
-                    expression,
-                    BracketedArgumentList(
+                var argumentList = BracketedArgumentList(
                         Token(SyntaxKind.OpenBracketToken).WithTriviaFrom(argList.OpenParenToken),
                         arguments,
-                        Token(SyntaxKind.CloseBracketToken).WithTriviaFrom(argList.CloseParenToken)));
+                        Token(SyntaxKind.CloseBracketToken).WithTriviaFrom(argList.CloseParenToken));
+                if (invocation.Expression is MemberBindingExpressionSyntax)
+                {
+                    return ElementBindingExpression(argumentList);
+                }
+
+                var expression = invocation.Expression is MemberAccessExpressionSyntax memberAccess
+                    ? memberAccess.Expression
+                    : invocation.Expression;
+                return ElementAccessExpression(expression, argumentList);
```

Comment:
nice, i like thsi a lot more.  my only suggestion (optional) is that in cases like this, we just show the before->After.  i.e. `x?.Substring(...) -> x?[...]` to help make it clear which user cases each is handling.  but it's not required here :)